Add end user and developer checkboxes (#3253)
quiltdata/quilt: docs/pull_request_template.md

- [ ] Unit tests
- [ ] Automated tests (e.g. Preflight)

> - [ ] Confirm this change does not bypass the SSO security model

Reviewer: ```suggestion
- [ ] Confirm that this change does not bypass the security model or best practices
```
We're going to need, separately, to document these and can include "SSO" in that list.

- [ ] Documentation
    - [ ] [Python: Run `build.py`](../gendocs/build.py) for new docstrings
    - [ ] JavaScript: basic explanation and screenshot of new features

>     - [ ] Text targeting end users (including first run)

Reviewer: ```suggestion
    - [ ] Markdown somewhere in docs/**/*.md that explains the feature to end users (said .md files should be linked from SUMMARY.md so they appear on https://docs.quiltdata.com)
```

>     - [ ] Text targeting developers (if relevant)

Reviewer: ```suggestion
    - [ ] Markdown docs for developers
```

"if relevant" not needed since authors can and should `[n/a]` bullets that don't apply.

Author: Applied
